lib/cylc/task_message.py

```diff
@@ -15,261 +15,135 @@
-                    ("%s=%s\n" % (
-                        self.CYLC_JOB_EXIT_TIME, self.true_event_time)))
-            elif message.startswith(self.ABORT_MESSAGE_PREFIX):
-                job_status_file.write(
-                    ("%s=%s\n" % (
-                        self.CYLC_JOB_EXIT,
-                        message[len(self.ABORT_MESSAGE_PREFIX):])) +
-                    ("%s=%s\n" % (
-                        self.CYLC_JOB_EXIT_TIME, self.true_event_time)))
-            elif message.startswith(self.VACATION_MESSAGE_PREFIX):
-                # Job vacated, remove entries related to current job
-                job_status_file_name = job_status_file.name
-                job_status_file.close()
-                lines = []
-                for line in open(job_status_file_name):
-                    if not line.startswith("CYLC_JOB_"):
-                        lines.append(line)
-                job_status_file = open(job_status_file_name, "wb")
-                for line in lines:
-                    job_status_file.write(line)
-                job_status_file.write("%s=%s|%s|%s\n" % (
-                    self.CYLC_MESSAGE, self.true_event_time, self.severity,
-                    message))
-            else:
-                job_status_file.write("%s=%s|%s|%s\n" % (
-                    self.CYLC_MESSAGE, self.true_event_time, self.severity,
-                    message))
-        try:
+            handle = sys.stdout
+        handle.write('%s %s - %s\n' % (event_time, severity, message))
+    handle.flush()
+    # Write to job.status
+    _append_job_status_file(suite, task_job, event_time, messages)
+    # Send messages
+    client = SuiteRuntimeServiceClient(suite)
+    try:
+        client.put_messages({
+            'task_job': task_job,
+            'event_time': event_time,
+            'messages': messages})
+    except ClientInfoError:
+        # Backward communication not possible
+        if cylc.flags.debug:
+            import traceback
+            traceback.print_exc()
+
+
+def _append_job_status_file(suite, task_job, event_time, messages):
+    """Write messages to job status file."""
+    job_log_name = os.getenv('CYLC_TASK_LOG_ROOT')
+    if not job_log_name:
+        job_log_name = os.path.join(
+            glbl_cfg().get_derived_host_item(suite, 'suite job log directory'),
+            'job')
+    job_status_file = None
+    try:
+        job_status_file = open(job_log_name + '.status', 'ab')
+    except IOError:
+        if cylc.flags.debug:
+            import traceback
+            traceback.print_exc()
+        return
+    for severity, message in messages:
+        if message == TASK_OUTPUT_STARTED:
+            job_id = os.getppid()
+            if job_id > 1:
+                # If os.getppid() returns 1, the original job process
+                # is likely killed already
+                job_status_file.write('%s=%s\n' % (CYLC_JOB_PID, job_id))
```

Comment:
I think it is actually clearer in the context here to use a string literal for format strings instead of introducing a constant.

Reply:
Fair enough! I was mainly thinking it would make it much simpler if we ever wanted to adjust the 'template' but would probably make the code a little less clear, yes.